core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -572,6 +556,19 @@ public static boolean assertNotClusterStateUpdateThread(String reason) {
         return true;
     }
 
+    /** asserts that the current stack trace does <b>NOT</b> invlove a cluster state applier */
+    private static boolean assertNotCalledFromClusterStateApplier(String reason) {
```

Comment:
yeah, I agree that we can address this later and discuss it when addressing #22108.